compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -10259,7 +10259,13 @@ TR::Node *ishlSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    if (identity)
       return identity;
 
-   if (secondChild->getOpCode().isLoadConst() &&
+   // Replace shift of constant zero with constant zero
+   if (firstChild->getOpCode().isLoadConst() && firstChild->getInt() == 0 &&
+       performTransformation(s->comp(), "%sRemoved redundant ishl node [%s]\n", s->optDetailString(), node->getName(s->getDebug())))
+      {
+      return s->replaceNode(node, firstChild, s->_curTree);
+      }
```

Comment:
It looks like the call to [`OMR::Optimization::replaceNode`](https://github.com/eclipse/omr/blob/b6e8df31bc3968bc1223723aa4399ff6c6ec696d/compiler/optimizer/OMROptimization.cpp#L511-L522) actually begins with a call to `performTransformation`, so the extra call to `performTransformation` can be removed.

Same comment will apply to the other calls to `replaceNode` guarded by `performTransformation` for both Simplifier and Value Propagation.